packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -106,8 +105,35 @@ export const ColorInput = forwardRef<InputElement, ColorInputProps>(
 
     const hasSuffix = Boolean(suffix);
 
+    const handlePaste: ClipboardEventHandler<InputElement> = (e) => {
+      if (
+        !colorPickerRef.current ||
+        !colorInputRef.current ||
+        disabled ||
+        readOnly
+      ) {
+        return;
+      }
+
+      e.preventDefault();
+
+      const pastedText = e.clipboardData.getData('text/plain').trim();
+
+      if (!pastedText || !/^#[0-9A-F]{6}$/i.test(pastedText)) {
+        return;
+      }
+
+      colorPickerRef.current.value = pastedText;
+      colorInputRef.current.value = pastedText.replace('#', '');
+      colorPickerRef.current.dispatchEvent(
+        new Event('change', { bubbles: true }),
+      );
+    };
+
     const onPickerColorChange: ChangeEventHandler<InputElement> = (e) => {
-      setCurrentColor(e.target.value);
+      if (colorInputRef.current) {
+        colorInputRef.current.value = e.target.value.replace('#', '');
```

Comment:
Might be worth to extract the formatting logic for the color input and color picker values into reusable functions.

Reply:
Tried and honestly didn't find it more readable. I think with just `#` even these inline calls should be understandable as long as all of them happen inside `ColorInput`. Happy to adjust though if you have specific suggestion how.